app/src/main/java/com/kickstarter/ui/activities/ResetPasswordActivity.kt

```diff
@@ -49,7 +50,10 @@ class ResetPasswordActivity : BaseActivity<ResetPasswordViewModel.ViewModel>() {
         this.viewModel.outputs.resetError()
                 .compose(bindToLifecycle())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe { ViewUtils.showDialog(this, getString(this.errorTitleString), getString(this.errorMessageString)) }
+                .subscribe {
+                    val message: Int = if( it == true ) this.errorNetworkMessageString
```

Comment:
We need to avoid any logic in the Activities. Any bussines logic should live on the viewmodel in order to test it.
What I mean here is that you should emit from the ViewModel directly the string you wanna display on the dialog

Reply:
+1 yes this logic should be moved into the view model. We do not have tests for activities but our viewmodels are backed by tests.